server/controllers/User.js

```diff
@@ -306,16 +331,94 @@ class UserController {
     return User.findOne({ where: { id: resetToken.id } })
       .then((user) => {
         if (user) {
-          user.update({ hash: hashPassowrd(password, user.salt) })
+          return user.update({ hash: hashPassword(password, user.salt) })
             .then(result => res.status(200)
               .json({
                 username: result.username, message: 'Password updated!',
               }));
-        } else {
-          return res.status(400)
+        }
+        return res.status(400)
+          .json({ errors: { message: 'User not found' } });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Get a profile
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getProfile(req, res, next) {
+    const { username } = req.params;
+    let owner = false;
+    const token = req.headers.authorization;
+    const decoded = token
+      ? tokenService.verifyToken(token) : { username: undefined };
+    // check to see if the profile belongs to the logged in user
+    if (decoded.username === username) owner = true;
+    User.findOne({ where: { username } })
+      .then((user) => {
+        if (!user) {
+          return res.status(404)
             .json({ errors: { message: 'User not found' } });
         }
-        return null;
+        return res.status(200).json({
+          profile: {
+            firstName: user.firstName,
+            lastName: user.lastName,
+            email: user.email,
+            username: user.username,
+            image: user.image,
+            bio: user.bio,
+            verified: user.verified,
+            owner,
+          },
+        });
+      })
+      .catch(next);
+  }
+
+
+  /**
+   * get all users, with the ability to search for users/profiles
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getAllProfile(req, res, next) {
+    // get query params
+    const { q } = req.query;
+    const limit = req.query.limit <= 20 ? req.query.limit : 20;
```

Comment:
Are you trying to limit maximum search rows to 20? If yes, why? I suggest you set a default value when no value is provided and allow the freedom to search more than 20 if desired. If you must set a max value it is usually more than 20.

Reply:
What number do you suggest for the maximum value?